src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/util/AzureStorageHelper.java

```diff
@@ -0,0 +1,125 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.util;
+
+import com.microsoft.azure.storage.blob.BlobAccessConditions;
+import com.microsoft.azure.storage.blob.BlockBlobURL;
+import com.microsoft.azure.storage.blob.ContainerURL;
+import com.microsoft.azure.storage.blob.ListBlobsOptions;
+import com.microsoft.azure.storage.blob.ListContainersOptions;
+import com.microsoft.azure.storage.blob.ReliableDownloadOptions;
+import com.microsoft.azure.storage.blob.ServiceURL;
+import com.microsoft.azure.storage.blob.models.BlobFlatListSegment;
+import com.microsoft.azure.storage.blob.models.BlobItem;
+import com.microsoft.azure.storage.blob.models.ContainerListBlobFlatSegmentResponse;
+import com.microsoft.azure.storage.blob.models.LeaseAccessConditions;
+import com.microsoft.azure.storage.blob.models.LeaseStateType;
+import com.microsoft.azure.storage.blob.models.ServiceListContainersSegmentResponse;
+import com.microsoft.rest.v2.Context;
+import com.microsoft.rest.v2.util.FlowableUtil;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.nio.ByteBuffer;
+
+@Component
+public class AzureStorageHelper {
+
+    private final ServiceURL client;
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureStorageHelper.class);
+
+    @Autowired
+    public AzureStorageHelper(ServiceURL client) {
+        this.client = client;
+    }
+
+    public ServiceURL getClient() {
+        return client;
+    }
+
+    public Single<ServiceListContainersSegmentResponse> listContainers() {
```

Comment:
👍 unfortunately once you start with rx everything will return a Rx object at some point.